storage/range_test.go

```diff
@@ -2411,28 +2410,33 @@ func TestRangeDanglingMetaIntent(t *testing.T) {
 		t.Errorf("expected original descriptor %s; got %s", &origDesc, &rlReply.Ranges[0])
 	}
 
-	// Try 100 lookups with IgnoreIntents. Expect roughly 50/50.
+	// Try 100 lookups with IgnoreIntents. Expect to see each descriptor at least once.
 	rlArgs.IgnoreIntents = true
-	var origCount, newCount int
-	for i := 0; i < 100; i++ {
-		clonedRLArgs := gogoproto.Clone(rlArgs).(*proto.InternalRangeLookupRequest)
-		clonedRLArgs.Timestamp = proto.ZeroTimestamp
-		rlReply.Reset()
+	var origSeen, newSeen bool
+	timeout := 50 * time.Millisecond
 
-		if err = tc.rng.AddCmd(tc.rng.context(), proto.Call{Args: clonedRLArgs, Reply: rlReply}, true); err != nil {
-			t.Fatal(err)
-		}
-		if reflect.DeepEqual(rlReply.Ranges[0], origDesc) {
-			origCount++
-		} else if reflect.DeepEqual(rlReply.Ranges[0], newDesc) {
-			newCount++
-		} else {
-			t.Errorf("expected orig/new descriptor %s/%s; got %s", &origDesc, &newDesc, &rlReply.Ranges[0])
+	for !(origSeen && newSeen) {
+		select {
+		case <-time.After(timeout):
```

Comment:
Don't you want to start the clock by creating the time.After() outside the loop? This will never trigger since the default will be chosen rather than the newly-created time.After.

I think it would be better, though, to limit the number of attempts directly instead of "as many requests as we can try in 50ms".